### PR TITLE
[AWS] Support VPC flow logs with message field

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
## What does this PR do?

When VPC flow logs sent directly from VPC without CloudWatch, they come in with the message field that looks like this:
```
{\"message\":\"2 428961148399 eni-0e0bf7be352692297 - - - - - - - 1671029698 1671029728 - NODATA\"}
```
This PR is to add support in the VPC flow log ingest pipeline to support this format.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

**Comment by andrewkroh** [comment]
Could you please elaborate in the PR description about what scenarios lead to this format. What writes the logs in the format and what input is used to read them?

**Comment by kaiyan-sheng** [comment]
@andrewkroh Please refer to the private github issue that I linked with the PR for more information. Thank you!

File: packages/aws/data_stream/vpcflow/_dev/test/pipeline/test-with-message-field.log [hunk 1]
```diff
@@ -0,0 +1 @@
+{\"message\":\"2 428961148399 eni-0e0bf7be352692297 - - - - - - - 1671029698 1671029728 - NODATA\"}
```
Comment (tommyers-elastic) [inline packages/aws/data_stream/vpcflow/_dev/test/pipeline/test-with-message-field.log:1]: is this sufficient to test the case where the document contains a top-level message field which contains a list? e.g. 

```
  "message": [
      "{\"message\":\"2 493751148399 eni-0e0bf7be782696797 - - - - - - - 1671324696 1671429728 - NODATA\"}\n"
    ],
```

side note: do we know _why_ these events look like this? could there ever be a situation where there are multiple objects in this list? 

thanks!

Comment (kaiyan-sheng) [reply packages/aws/data_stream/vpcflow/_dev/test/pipeline/test-with-message-field.log:1]: I think the test case is efficient, I don't think the nested message level actually happens hmmm

The events look like this when VPC flow logs are sent directly without going through CloudWatch.

**Review by tommyers-elastic — APPROVED** [review]
LGTM nice work kaiyan ! is there a way to test the dot expander change?

**Comment by kaiyan-sheng** [comment]
@tommyers-elastic Thanks. I didn't find a way to add test into for example `_dev/test/pipeline` but we can test it manually using the Kibana dev tools simulate. For example:
```
POST _ingest/pipeline/logs-aws.vpcflow-1.28.2/_simulate
{
  "docs": [
    {
      "_source": {
        "message": "2 428152502467 eni-0c1c3a5c0ac2c4c95 162.142.125.229 172.31.29.121 5583 6352 6 1 44 1648766030 1648766085 REJECT OK",
        "@timestamp": "2022-03-31T22:33:50Z",
        "event.id": "36768711127022897352489210216814512733652448397667467264",
        "cloud.provider": "aws",
        "cloud.account.id": "428152502467",
        "cloud.region": "us-east-1",
        "aws.firehose.arn": "arn:aws:firehose:us-east-1:428152502467:deliverystream/vpc-flow-log-stream-http-endpoint",
        "aws.firehose.request_id": "a70eae51-0629-4c78-b0dc-799a19642585",
        "aws.kinesis.type": "deliverystream",
        "aws.kinesis.name": "vpc-flow-log-stream-http-endpoint"
      }
    }
    ]
}
```
Without the `dot_expander` processor, the output will have `cloud.provider` twice, one flat and one nested.

**Pull request merged**